src/organizations/components/OrgProfileTab/index.tsx

```diff
@@ -24,13 +24,30 @@ import {CLOUD} from 'src/shared/constants'
 import {getMe} from 'src/me/selectors'
 
 import 'src/organizations/components/OrgProfileTab/style.scss'
+import {
+  getBillingProviderThunk,
+  getCurrentOrgDetailsThunk,
+} from 'src/identity/actions/thunks'
+import {shouldUseQuartzIdentity} from 'src/identity/utils/shouldUseQuartzIdentity'
 
 const OrgProfileTab: FC = () => {
   const me = useSelector(getMe)
   const org = useSelector(getOrg)
+  const dispatch = useDispatch()
 
   const expectQuartzData = CLOUD && isFlagEnabled('uiUnificationFlag')
 
+  useEffect(() => {
+    if (CLOUD && shouldUseQuartzIdentity()) {
+      if (!me.quartzMe.billingProvider) {
+        dispatch(getBillingProviderThunk())
+      }
+      if (CLOUD && !me.quartzMe.regionCode) {
```

Comment:
No need for this `CLOUD` check here, since the outside `if` statement already handles that.

In response to the question I posed in the identity thunks regarding whether the conversion from `identity` to `quartzMe` needs to happen - it doesn't seem like it, since if this code gets run, then `shouldUseQuartzIdentity()` is true, and the fetching of `identity` has already happened, since that's one of the first actions the app performs. Is that a correct or incorrect assessment?

Reply:
Good point - removed the `CLOUD` check that is already happening.

Re: other question, it's correct that if this code is running, we're using `quartzIdentity`. But see earlier discussion re: need to convert to legacy `me` state any time `identity` is updated, since they need to stay in sync, and since the selectors in the app pull from `quartz/me` whether the flag is on or not.